src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/scarlet_monastery/boss_mograine_and_whitemane.cpp

```diff
@@ -173,230 +212,303 @@ struct boss_scarlet_commander_mograineAI : public ScriptedAI
+        ResetCombatAction(WHITEMANE_ACTION_POWERWORD_SHIELD, GetSubsequentActionTimer(WHITEMANE_ACTION_POWERWORD_SHIELD));
+        ResetCombatAction(WHITEMANE_ACTION_HOLY_SMITE, GetSubsequentActionTimer(WHITEMANE_ACTION_HOLY_SMITE));
+        ResetCombatAction(WHITEMANE_ACTION_DOMINATE_MIND, GetSubsequentActionTimer(WHITEMANE_ACTION_DOMINATE_MIND));
+        SetCombatScriptStatus(false);
+        SetCombatMovement(true);
+        SetMeleeEnabled(true);
+    }
+
+    void HandleSalute()
+    {
+        m_creature->HandleEmote(EMOTE_ONESHOT_SALUTE);
+    }
 
-        // m_uiHolySmite_Timer
-        if (m_uiHolySmite_Timer < uiDiff)
+    void ExecuteAction(uint32 action) override
+    {
+        switch (action)
         {
-            if (DoCastSpellIfCan(m_creature->GetVictim(), SPELL_HOLYSMITE) == CAST_OK)
-                m_uiHolySmite_Timer = urand(3500, 5000);
+            case WHITEMANE_ACTION_HEAL:
+            {
+                if (Unit* target = DoSelectLowestHpFriendly(50.0f))
+                {
+                    if (DoCastSpellIfCan(target, SPELL_HEAL) == CAST_OK)
+                        ResetCombatAction(action, GetSubsequentActionTimer(action));
+                }
+                return;
+            }
+            case WHITEMANE_ACTION_HOLY_SMITE:
+            {
+                if (DoCastSpellIfCan(m_creature->GetVictim(), SPELL_HOLYSMITE) == CAST_OK)
+                    ResetCombatAction(action, GetSubsequentActionTimer(action));
+                return;
+            }
+            case WHITEMANE_ACTION_POWERWORD_SHIELD:
+            {
+                if (DoCastSpellIfCan(nullptr, SPELL_POWERWORDSHIELD) == CAST_OK)
+                    ResetCombatAction(action, GetSubsequentActionTimer(action));
+                return;
+            }
+            case WHITEMANE_ACTION_DEEP_SLEEP:
+            {
+                if (m_creature->GetHealthPercent() <= 50.0f)
+                {
+                    DoCastSpellIfCan(nullptr, SPELL_DEEPSLEEP, CAST_INTERRUPT_PREVIOUS);
+                    SetCombatMovement(false);
+                    SetMeleeEnabled(false);
+                    SetCombatScriptStatus(true);
+                    m_creature->SetTarget(nullptr);
+                    if (Creature* mograine = m_instance->GetSingleCreatureFromStorage(NPC_MOGRAINE))
+                    {
+                        SetCombatMovement(false);
+                        m_creature->AttackStop(true);
+                        float fX, fY, fZ;
+                        mograine->GetContactPoint(m_creature, fX, fY, fZ, INTERACTION_DISTANCE / 2.f);
+                        m_creature->GetMotionMaster()->Clear();
+                        m_creature->GetMotionMaster()->MovePoint(2, fX, fY, fZ, FORCED_MOVEMENT_RUN);
+                    }
+                    DisableCombatAction(action);
+                }
+                return;
+            }
+            case WHITEMANE_ACTION_DOMINATE_MIND:
+            {
+                if (!urand(0, 10))
```

Comment:
this seems way too likely given my own tests on reference and videos I watched of the reference